Library/Homebrew/test.rb

```diff
@@ -17,6 +17,15 @@
   ENV.extend(Stdenv)
   ENV.setup_build_environment
 
+  # This ensures all executables in test do are called with a full path, i.e.
+  # bin/"wget", or Formula["openssl"].opt_bin/"openssl" rather than just "wget".
+  # It's very likely to break tests, so phase it in via CI rather than user
```

Comment:
Also your change has the potential to create false negatives when the brew paths are excised such that the utility called ends up being something else (likely a system binary or a non-brew user binary with the same name), and the something else happens to pass the test. So had you not intervened, the test would have correctly failed with a true positive, but because it fell through to the other binary it "silently" passes with a false negative.